utils/python/CIME/user_mod_support.py

```diff
@@ -56,6 +56,8 @@ def apply_user_mods(caseroot, user_mods_path, ninst=None):
                             expect(False, "Could not write file %s in caseroot %s"
                                    %(case_source_mods,caseroot))
 
+    include_dirs.reverse()
```

Comment:
I did test it, that's the proper use for `.reverse()`; to do it inline it would be `for include_dir in include_dirs[::-1]:`

which I didn't like as much.